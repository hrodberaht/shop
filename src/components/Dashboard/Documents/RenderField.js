import React from 'react';
import ShowErrorInForm from './ShowErrorInForm';

const RenderField = ({ input, label, type, meta }) => (
  <div>
    <label htmlFor={label}>{label}</label>
    <input className="form-input" {...input} type={type} />
    <ShowErrorInForm className="error-text" {...meta} />
  </div>
);

export default RenderField;
