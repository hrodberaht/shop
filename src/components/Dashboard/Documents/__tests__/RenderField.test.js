import RenderField from '../RenderField';

const defaultProps = {
  label: 'Company'
};

const setup = buildSetup(RenderField, defaultProps);

describe('<RenderField />', () => {
  describe('when initialized', () => {
    it('should render without crash', () => {
      const { wrapper } = setup();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
