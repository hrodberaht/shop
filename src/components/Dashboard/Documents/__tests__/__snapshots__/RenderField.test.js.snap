// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RenderField /> when initialized should render without crash 1`] = `
<div>
  <label
    htmlFor="Company"
  >
    Company
  </label>
  <input
    className="form-input"
  />
  <ShowErrorInForm
    className="error-text"
    error={null}
    touched={false}
  />
</div>
`;
